server/src/test/java/org/elasticsearch/snapshots/SnapshotsInProgressSerializationTests.java

```diff
@@ -147,7 +147,7 @@ protected Custom makeTestChanges(Custom testInstance) {
                 for (int i = 0; i < entries.size(); i++) {
                     if (randomBoolean()) {
                         final Entry entry = entries.get(i);
-                        entries.set(i, mutateEntry(entry));
+                        entries.set(i, mutateEntryWithLegalChange(entry));
                     }
                 }
                 updatedInstance = updatedInstance.withUpdatedEntriesForRepo(perRepoEntries.get(0).repository(), entries);
```

Comment:
Can we also reorder the entries for one or more repositories? Really just to give the `ByRepoDiff` stuff a proper workout.

Reply:
Jup added shuffling the list. Had to change the way we generate the repo name for that. It was a random string per snapshot and we'd effectively never see collisions there. Now we actually have multiple snapshots per repo here and the position reordering is covered by tests.

Reply:
:+1: sounds good